fw/stat.h

```diff
@@ -0,0 +1,111 @@
+/**
+ *		Tempesta kernel library
+ *
+ * Copyright (C) 2016-2024 Tempesta Technologies, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+#ifndef __TFW_STAT_H__
+#define __TFW_STAT_H__
+
+#include "http.h"
+
+/* Shared structures used for statistics. */
+
+/*
+ * Structures for health monitoring statistics accountings in procfs.
+ *
+ * @code	- HTTP code for which the statistics is collected.
+ * @sum		- Amount of responses for some limited period.
+ * @total	- Total amount of responses for the overall period.
```

Comment:
I know this isn't your doing, but why not make it a bit better? Below are thoughts on both better naming and better comments.

First, it's unclear that these are structures for backend health monitoring which account only for responses from backends.

`HTTP code` - this is called HTTP response code, or - better, in my mind - HTTP status code. It's often called just `status`, aka response status code. I believe `status` is more identifiable than just `code`. But `code` is quite just okay if it's aligned with other names on the structure.

`sum` and `total` look like synonyms, and thus difficult to remember which is which. Perhaps, `sum` can be `nresp` with the comment `Number of responses in a specified time frame`. Other names are welcome (like `rcnt`?), anything is better than inexpressive `sum`! Then, `total` looks fine, possible with a better comment `Total number of responses` (total is total, saying that it's _overall_ is meaningless).

Alternatives, as this is all about responses with specific status, could be: `status`, `count`, `total`. Or `code`, `count`, `total`. Or `rcode`, `rcount`, `rtotal` where `r` means `response`. When confined within `CodeStats{}` structure, these names have clear meaning.

Reply:
I don't know... We have a lot of such things around the project. I just skipped them before to not inflate the PRs, considering it as minor refactoring.

In my opinion, sum, nresp, rcnt - it's all the same, no additional information. We should use tf_total (tf=time frame), for example, if we want to add some information. It's quite subjective thing I think.

Reply:
I want to discuss it on meeting to form a common idea about that topic, I suppose it can be helpful.

Reply:
Oh, it's definitely subjective! I just presented my opinion which is based on my knowledge of English and the proper terminology. :-) Usually I try to give variables names that are meaningful in the context to anyone that knows the terminology, or just follow the common sense. I was difficult for me to follow this code as every time I saw these names I had to get back and read their description again. As for `tf_total`, a time frame is something partial by definition, and having the total within a partial feels strange. Nevertheless, as they say _the best is the enemy of the good_.

And of course it doesn't have to be a part of this PR.